kortex_driver/src/hardware_interface.cpp

```diff
@@ -758,8 +758,21 @@ return_type KortexMultiInterfaceHardware::read(
 {
   if (first_pass_)
   {
+    // Wait here to mitigate exceptions on startup with low compute devices  
+    std::this_thread::sleep_for(std::chrono::milliseconds(500));
     first_pass_ = false;
-    feedback_ = base_cyclic_.RefreshFeedback();
+    try
+    {
+      feedback_ = base_cyclic_.RefreshFeedback();
+    }
+    catch (k_api::KDetailedException & ex)
+    {
+      RCLCPP_ERROR_STREAM(LOGGER, "Kortex exception during read first pass: " << ex.what());
+
+      RCLCPP_ERROR_STREAM(
+      LOGGER, "Error sub-code: " << k_api::SubErrorCodes_Name(
+       	         k_api::SubErrorCodes((ex.getErrorInfo().getError().error_sub_code()))));
+    }
```

Comment:
A little late to the party, sorry. 
So my understanding is that this happens on startup but I think it may be a cleaner solution to put this in `on_activate()`, right? @MarqRazz would that fit well with the driver?

Reply:
I could try it in activate

Reply:
If we were to move `base_cyclic_.Refresh()` and `base_cyclic_.RefreshFeedback()` to their own functions it would keep us from repeating the try-catch statement every time we call them.

Reply:
yeah sounds like a good idea